pulsar-client-auth-athenz/src/main/java/org/apache/pulsar/client/impl/auth/AuthenticationAthenz.java

```diff
@@ -81,23 +82,24 @@ public String getAuthMethodName() {
     }
 
     @Override
-    synchronized public AuthenticationDataProvider getAuthData() throws PulsarClientException {
-        if (cachedRoleTokenIsValid()) {
-            return new AuthenticationDataAthenz(roleToken, isNotBlank(roleHeader) ? roleHeader : ZTSClient.getHeader());
-        }
-        try {
-            // the following would set up the API call that requests tokens from the server
-            // that can only be used if they are 10 minutes from expiration and last twenty
-            // four hours
-            RoleToken token;
-            synchronized (providerDomainLock) {
-                token = getZtsClient().getRoleToken(providerDomain, null, minValidity, maxValidity, false);
+    public AuthenticationDataProvider getAuthData() throws PulsarClientException {
+        synchronized (cachedRoleTokenLock) {
```

Comment:
This could actually be converted into a read-write lock, taking the read lock and upgrading to write lock to refresh the token

Reply:
@merlimat That's right. Introduced read-write lock.